SPECS-EXTENDED/rubygem-hpricot/rubygem-hpricot.spec

```diff
@@ -0,0 +1,356 @@
+%define	gem_name		hpricot
+
+# See bug 1402582
+# Currently colm (dependency for ragel) has difficulty on
+# armv7hl
+%global	force_regenerate_from_rl	0
+
+Summary:	A Fast, Enjoyable HTML Parser for Ruby
+Name:		rubygem-%{gem_name}
+Version:	0.8.6
+Release:	34%{?dist}
+# ext/fast_xs/FastXsService.java is licensed under ASL 2.0
+License:	MIT AND ASL 2.0
+Vendor:	        Microsoft Corporation
+Distribution:	Mariner
+URL:		http://github.com/hpricot/hpricot
+# Non-free file removed, see Source10
+# Source0:	http://gems.rubyforge.org/gems/%{gem_name}-%{version}.gem
+Source0:	https://src.fedoraproject.org/repo/pkgs/rubygem-hpricot/hpricot-0.8.6-modified.gem/5d737234720f3847cc5def6c0047f2e4/hpricot-0.8.6-modified.gem
+Source10:	rubygem-hpricot-create-free-gem.sh
+
+Requires:	ruby(release)
+BuildRequires:	ruby(release)
+
+BuildRequires:	gcc
+BuildRequires:	rubygems-devel
+# Recompile
+%if %{force_regenerate_from_rl}
+BuildRequires:	ruby
+BuildRequires:	ragel
+%endif
+# Others
+BuildRequires:	rubygem(rake-compiler)
+BuildRequires:	ruby-devel
+BuildRequires:  rubygem(test-unit)
+BuildRequires:  ruby
+Requires:	ruby(rubygems)
+Provides:	rubygem(%{gem_name}) = %{version}-%{release}
+%if 0%{?fedora} >= 17
+Obsoletes:	ruby-%{gem_name} <= %{version}-%{release}
+Provides:	ruby-%{gem_name} = %{version}-%{release}
+%endif
+
+%description
+Hpricot is a very flexible HTML parser, based on Tanaka Akira's
+HTree and John Resig's JQuery, but with the scanner recoded in C
+(using Ragel for scanning.)
+
+%package	doc
+Summary:	Documentation for %{name}
+# Directory ownership issue
+Requires:	%{name} = %{version}-%{release}
+Requires:	ruby(rubygems)
+
+%description	doc
+This package contains documentation for %{name}.
+
+%package	-n ruby-%{gem_name}
+Summary:	Non-Gem support package for %{gem_name}
+Requires:	%{name} = %{version}-%{release}
+Provides:	ruby(%{gem_name}) = %{version}-%{release}
+
+%description	-n ruby-%{gem_name}
+This package provides non-Gem support for %{gem_name}.
+
+%prep
+%setup -q -T -c
+
+# Gem repack
+TOPDIR=$(pwd)
+mkdir tmpunpackdir
+pushd tmpunpackdir
+
+gem unpack %{SOURCE0}
+cd %{gem_name}-%{version}*
+
+# ??
+find . -type f | xargs chmod ugo+r
+
+sed -i Rakefile -e '\@require.*bundler/setup@d'
+mkdir tmpbin
+ln -sf /bin/true tmpbin/git
+export PATH=$(pwd)/tmpbin:$PATH
+
+%if %{force_regenerate_from_rl}
+rm -f ext/hpricot_scan/*.c
+rake ext/hpricot_scan/extconf.rb
+%endif
+
+gem specification -l --ruby %{SOURCE0} > %{gem_name}.gemspec
+gem build %{gem_name}.gemspec
+mv %{gem_name}-%{version}.gem $TOPDIR
+
+popd
+
+%gem_install
+
+pushd .%{gem_instdir}/test
+# Kill tests related to BOINGBOING, licensed under CC-BY-NC
+grep -rl BOING . | \
+	xargs sed -i '/BOING/s|^\([ \t][ \t]*\)\(.*\)$|\1# This test is intentionally killed\n\1return true\n\1\2|'
+popd
+
+%build
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a ./%{gem_dir}/* %{buildroot}%{gem_dir}
+
+%if 0%{?fedora} >= 21 || 0%{?rhel} > 7
```

Comment:
Please remove macros referencing to Fedora or RedHat. They make no sense in Mariner.

Reply:
fixed

Reply:
I think you removed the wrong branch for the `%if 0%{?fedora} >= 21 || 0%{?rhel} > 7` statement.

Reply:
Discussed offline, looks good, thanks!